proton-j/src/main/java/org/apache/qpid/proton/message/impl/MessageImpl.java

```diff
@@ -666,9 +784,9 @@ public void decode(ReadableBuffer buffer)
             }
 
         }
-        if(section != null && !(section instanceof Footer))
+        while(section != null && !(section instanceof Footer))
         {
-            _body = section;
+            addBodySection(section);
```

Comment:
This should fail if receiving multiple body sections with differing types (or multiple AmqpValue sections)